pymove/tests/preprocessing/test_filters.py

```diff
@@ -0,0 +1,567 @@
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1]
+    ], columns=cols, index=[0, 1]
+    )
+    assert_frame_equal(filter_values, expected)
+    assert move_df.len() == 4
+
+    filter_out_values = filters.by_id(move_df, id_=1, filter_out=True)
+    expected = DataFrame([
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[2, 3]
+    )
+    assert_frame_equal(filter_out_values, expected)
+    assert move_df.len() == 4
+
+    filters.by_id(move_df, id_=1, filter_out=True, inplace=True)
+    expected = DataFrame([
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[2, 3]
+    )
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 2
+
+
+def test_by_tid():
+    move_df = MoveDataFrame(
+        data=list_data_1, latitude='lat', longitude='lon', datetime='datetime',
+        traj_id='id'
+    )
+    move_df['tid'] = ['123', '456', '789', '789']
+    cols = ['lat', 'lon', 'datetime', 'id', 'tid']
+
+    filter_values = filters.by_tid(move_df, tid_='456')
+    expected = DataFrame([
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1,
+         '456'],
+    ], columns=cols, index=[1]
+    )
+    assert_frame_equal(filter_values, expected)
+    assert move_df.len() == 4
+
+    filter_out_values = filters.by_tid(move_df, tid_='456', filter_out=True)
+    expected = DataFrame([
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1,
+         '123'],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2,
+         '789'],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2,
+         '789']
+    ], columns=cols, index=[0, 2, 3]
+    )
+    assert_frame_equal(filter_out_values, expected)
+    assert move_df.len() == 4
+
+    filters.by_tid(move_df, tid_='789', filter_out=False, inplace=True)
+    expected = DataFrame([
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2,
+         '789'],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2,
+         '789']
+    ], columns=cols, index=[2, 3]
+    )
+
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 2
+
+
+def test_outliers():
+    move_df, cols = _prepare_df_with_distances()
+
+    outliers = filters.outliers(move_data=move_df, jump_coefficient=1)
+    expected = DataFrame(data=[
+        [1, 38.984211, 115.319389, Timestamp('2008-10-23 05:53:11'), 140454.64510608764,
+         140460.97747220137, 7.563335999941849],
+    ], columns=cols, index=[2]
+    )
+    assert_frame_equal(outliers, expected)
+    assert move_df.len() == 5
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574862)